packages/client/hmi-client/src/services/model.ts

```diff
@@ -173,6 +173,16 @@ export async function getModelEquation(model: Model): Promise<string> {
 	return latex ?? '';
 }
 
+export const getUnitsFromModelParts = (model: Model) => {
+	const modelVariableUnits: { [key: string]: string } = {
+		_time: model?.semantics?.ode?.time?.units?.expression || ''
+	};
+	[...(model?.model.states ?? []), ...(model?.semantics?.ode?.parameters ?? [])].forEach((v) => {
+		modelVariableUnits[v.id] = v.units?.expression || '';
+	});
+	return modelVariableUnits;
+};
```

Comment:
If we do it this way, to pre-calculate the units per variable - we also need to compute for
- observables (which has no expression but their children do), and
- time

Reply:
I updated it so that it grabs the unit for the observable from the first state in `observables.states`. I wonder if there are cases where the units from states are different and how we should handle in such case.